plan/config.go

```diff
@@ -104,10 +114,54 @@ func parseBehavior(d string) Behavior {
 		ClientAxis: clientAxis,
 		ParamsAxes: values,
 	}
-
 	return behavior
 }
 
+func validateAndApplyFilters(filterMap map[string][]Filter, behaviors Behaviors) (Behaviors, error) {
+	for i := range behaviors {
+		behavior := &behaviors[i]
+		filters := filterMap[behavior.Name]
+		for _, filter := range filters {
+			for match := range filter.AxisMatches {
+				if match != behavior.ClientAxis && !axisContainsRule(match, behavior.ParamsAxes) {
+					return nil, fmt.Errorf("%v is not defined in axis for %v", match, behavior.Name)
+				}
+			}
+		}
+		behavior.SkipFilters = filters
+	}
+	return behaviors, nil
+}
+
+func axisContainsRule(item string, axis []string) bool {
+	for _, axes := range axis {
+		if axes == item {
+			return true
+		}
+	}
+	return false
+}
+
+func parseSkipBehavior(d string) (string, []Filter) {
+	pair := strings.SplitN(d, "=", 2)
+	key := strings.ToLower(pair[0])
+	filters := strings.Split(pair[1], ",")
```

Comment:
Good point. I also saw we behaved in similar way up while parsing axis and behaviors. 

Yes, the renamed helped with readability. Thanks.